File: 1653-minimum-deletions-to-make-string-balanced/main.go
func minimumDeletions(s string) int {
    
    bCount, deletions := 0, 0
    
    for _, char := range s {
        if char == 'b'{ 
            bCount++
        }else {
            deletions = min(deletions + 1, bCount)
        }
    }

    return deletions
}

func min(a, b int) int  {
    if a > b {
        return b
    }
    return a
}

File: 1653-minimum-deletions-to-make-string-balanced/main.py
class Solution:
    def minimumDeletions(self, s: str) -> int:
        b_cnt, num_deletions = 0, 0

        for char in s:
            if char == 'b': b_cnt += 1
            else:
                num_deletions = min(num_deletions + 1, b_cnt)
            
        return num_deletions
